src/calculator.ts

```diff
@@ -1,54 +1,86 @@
 import fs from "fs";
 import { linearQF, Contribution, Calculation } from "pluralistic";
+import {AugmentedResult, CalculatorOptions, RawContribution} from "./types.js";
 
-type AugmentedResult = Calculation & {
-  projectName: string;
-  payoutAddress: string;
-};
-
-interface RawContribution {
-  voter: string;
-  projectId: string;
-  amountUSD: number;
-}
 
 export default class Calculator {
   private baseDataPath: string;
   private chainId: string;
   private roundId: string;
   private matchAmount: number;
+  private minimumAmount: number | undefined;
+  private passport: boolean | undefined;
+  private passportThreshold: number | undefined;
 
-  constructor(
-    baseDataPath: string,
-    chainId: string,
-    roundId: string,
-    matchAmount: number
-  ) {
+  constructor(options: CalculatorOptions) {
+    const {
+        baseDataPath,
+        chainId,
+        roundId,
+        matchAmount,
+        minimumAmount,
+        passport,
+        passportThreshold,
+    } = options;
     this.baseDataPath = baseDataPath;
     this.chainId = chainId;
     this.roundId = roundId;
     this.matchAmount = matchAmount;
+    this.minimumAmount = minimumAmount;
+    this.passport = passport;
+    this.passportThreshold = passportThreshold;
   }
 
   calculate() {
     const rawContributions = this.parseJSONFile(
-      `${this.chainId}/rounds/${this.roundId}/votes.json`
+        `${this.chainId}/rounds/${this.roundId}/votes.json`
     );
     const projects = this.parseJSONFile(`${this.chainId}/projects.json`);
     const applications = this.parseJSONFile(
-      `${this.chainId}/rounds/${this.roundId}/projects.json`
+        `${this.chainId}/rounds/${this.roundId}/projects.json`
+    );
+    const rounds = this.parseJSONFile(
+        `${this.chainId}/rounds.json`
     );
+    const passportScores = this.parseJSONFile("passport_scores.json");
+    const validAddresses = this.parseJSONFile("passport_valid_addresses.json");
 
-    const contributions: Array<Contribution> = rawContributions.map(
-      (raw: RawContribution) => ({
-        contributor: raw.voter,
-        recipient: raw.projectId,
-        amount: raw.amountUSD,
-      })
+    const currentRound = rounds.find((r: any) => r.id === this.roundId);
+    let contributions: Array<Contribution> = rawContributions.map(
+        (raw: RawContribution) => ({
+          contributor: raw.voter,
+          recipient: raw.projectId,
+          amount: raw.amountUSD,
+        })
     );
 
+    contributions = contributions.filter((c: Contribution) => {
+      const addressData = passportScores.find((ps: any) => ps.address === c.contributor);
+      const hasValidEvidence = addressData?.evidence?.success;
+      const isValidAddress = validAddresses.includes(c.contributor);
+
+      if (this.passport) {
+        if (typeof this.passportThreshold !== 'undefined') {
+          return (
+              c.amount >= (this.minimumAmount ?? currentRound.minimumAmount ?? 0) &&
```

Comment:
I see `this.minimumAmount ?? currentRound.minimumAmount` happens 4 times in this function, what if we do it in the constructor of the class? we store the right minimumAmount to use, or keep it like this but make a getter of some kind.

Reply:
`calculate` method has been refactored to reduce redundancy while preserving the original behavior. I moved the calculation of `minAmount` is moved outside the filter function, and a separate function `isEligible` is introduced to determine the eligibility of a contribution based on the passport and passport threshold params.